include/xtensor/xchunk_store_manager.hpp

```diff
@@ -84,23 +106,59 @@ namespace xt
                     path.append(".");
                 path.append(std::to_string(*it));
             }
-            m_file_array.set_path(path);
+            if (path.empty())
+            {
+                return m_file_array[0];
+            }
+            else
+            {
+                // check if the chunk is already loaded in memory
+                std::vector<std::string>::iterator it = std::find(m_path.begin(), m_path.end(), path);
+                size_t index;
+                if (it != m_path.end())
+                {
+                    index = std::distance(m_path.begin(), it);
+                    return m_file_array[index];
+                }
+                // if not, get a free chunk in the pool
+                index = 0;
+                bool free_chunk = false;
+                for (auto p: m_path)
+                {
+                    if (p.empty())
+                    {
+                        free_chunk = true;
+                        break;
+                    }
+                    index += 1;
+                }
+                if (free_chunk)
+                {
+                    m_file_array[index].set_path(path);
+                    m_path[index] = path;
+                    return m_file_array[index];
+                }
+                // no free chunk, take one (which will thus be unloaded)
+                // current algorithm takes one at random to be fair
```

Comment:
Instead of being random, we could use an `std::queue`, for a first-in / first-out strategy.

Reply:
The problem with `std::queue` or `std::deque` is that e.g. `push_back` creates a new element, which we don't want. We want all the elements to be created in the pool initialization, and then we just call `set_path` when loading a chunk. In the end, fairness is achieved through a simple walking index in the pool `std::vector`.